File: csharp/Chapter07/Address.cs

namespace com.db4o.dg2db4o.chapter7
{
    class Address
    {
        string _street;
        string _city;
        string _country;
        ZipCode _zipCode;
       // Customer _customer;

        public Address(string street, string city, string country)
        {
            _street = street;
            _city = city;
            _country = country;
        }

        public string Street
        {
            get
            {
                return _street;
            }
            set
            {
                _street = value;
            }
        }

        public string City
        {
            get
            {
                return _city;
            }
            set
            {
                _city = value;
            }
        }

        public string Country
        {
            get
            {
                return _country;
            }
            set
            {
                _country = value;
            }
        }

        public ZipCode ZipCode
        {
            get
            {
                return _zipCode;
            }
            set
            {
                _zipCode = value;
            }
        }

        //public Customer Customer
        //{
        //    get
        //    {
        //        return _customer;
        //    }
        //    set
        //    {
        //        _customer = value;
        //    }
        //}

        public override string ToString()
        {
            return _street + ", " + _city + ", " + _country + " (Address)";
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter7/Address.java

package com.db4o.dg2db4o.chapter7;

public class Address {
    
    String _street;
    String _city;
    String _country;
    private ZipCode _zipCode;
    Customer _customer;
    
    public Address(String street, String city, String country) {
        _street = street;
        _city = city;
        _country = country;
    }
    
    public String getStreet() {
        return _street;
    }
    public void setStreet(String value) {
        _street = value;
    }
    
    public String getCity() {
        return _city;
    }
    public void setCity(String value) {
        _city = value;
    }
    
    public String getCountry() {
        return _country;
    }
    public void setCountry(String value) {
        _country = value;
    }
    
    public Customer getCustomer() {
        return _customer;
    }
    public void setCustomer(Customer value) {
        _customer = value;
    }
    
    public ZipCode getZipCode() {
        return _zipCode;
    }

    public void setZipCode(ZipCode zipCode) {
        this._zipCode = zipCode;
    }
    
    public String toString() {
        return _street + ", " + _city + ", " + _country;
    }
  
}
